### PR TITLE
[Relay][TensorFlow] Remove 'input_0d_mismatch' special handling

### PR DESCRIPTION
**Pull request by lixiaoquan**
@srkreddy1238 @alexeyr Please review

### CODE REVIEW

**Comment by alexeyr** [comment]
I won't be able to review before going on vacation, though it's good if Relay doesn't need it anymore. Maybe add some tests directly for `tf.expand_dims` to https://github.com/dmlc/tvm/blob/master/tests/python/frontend/tensorflow/test_forward.py? There don't seem to be any at the moment.

**Comment by lixiaoquan** [comment]
Test is added

File: tests/python/frontend/tensorflow/test_forward.py [hunk 1]
```diff
@@ -1475,6 +1475,19 @@ def test_forward_rel_ops():
...
+def _test_forward_expand_dims(data, axis):
+    in1 = tf.placeholder(shape=data.shape, dtype=data.dtype, name='in1')
+    out = tf.expand_dims(in1, axis)
+    compare_tf_with_tvm([data], [in1.name], out.name)
+
+def test_forward_expand_dims():
+    _test_forward_expand_dims(np.int32(1), 0)
+    _test_forward_expand_dims(np.array([1]), 0)
+    _test_forward_expand_dims(np.array([[1], [2]]), 0)
+    _test_forward_expand_dims(np.array([[1], [2]]), 1)
```
Comment (yongwww) [inline tests/python/frontend/tensorflow/test_forward.py]: not sure if "axis = -1" is supported here

Comment (lixiaoquan) [reply tests/python/frontend/tensorflow/test_forward.py]: It's added

File: python/tvm/relay/frontend/tensorflow.py [hunk 2]
```diff
@@ -820,9 +819,9 @@ def _transform_mask(stride_dim, ellipsis_mask):
                 pass
             else:
                 final_output.append(out_shape[gather_index])
-        # Prevent 0-dim tensors which are not accepted by Relay
```
Comment (srkreddy1238) [inline python/tvm/relay/frontend/tensorflow.py]: Check if we have a test case to cover this else we should add one.

**Comment by yzhliu** [comment]
Thanks @lixiaoquan @srkreddy1238 @yongwww

**Pull request merged**